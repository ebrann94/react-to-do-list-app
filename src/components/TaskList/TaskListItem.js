import React from 'react';

const TaskListItem = ({ handleCompleteTask, handleDeleteTask, ...props }) => {
    const className = props.completed ? 'completed list-item' : 'list-item';
    
    return (
        <div 
            className={className} 
            onClick={(e) => {
                if (e.target.name !== 'delete-btn') handleCompleteTask(props.id)
            }}
        >   
            <p>{props.text}</p>
            <div className="list-item__btns">
                <button
                    name="delete-btn"
                    className="list-item__delete"
                    onClick={() => handleDeleteTask(props.id)}
                >
                    X
                </button>
            </div>
        </div>
    );
};

export default TaskListItem;