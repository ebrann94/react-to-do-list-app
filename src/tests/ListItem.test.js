import React from 'react';
import { shallow, mount } from 'enzyme';
import TaskListItem from '../components/TaskList/TaskListItem';

describe('TaskListItem component tests', () => {
    let wrapper;
    const props = {
        text: 'Example text',
        id: '1234',
        completed: false,
        handleCompleteItem: jest.fn(),
        handleRemoveOne: jest.fn()
    }

    beforeEach(() => {
        wrapper = shallow(<TaskListItem {...props} />)
    });

    afterEach(() => {
        wrapper.unmount();
    });

    it('renders without crashing given required props', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders text correctly', () => {
        expect(wrapper.find('p').text()).toBe(props.text);
    });

    it('calls complete item when clicked', () => {
        const event = {
            target: {
                id: '---'
            }
        }
        wrapper.find('div.list-item').simulate('click', event);
        expect(props.handleCompleteItem).toHaveBeenCalledWith(props.id);
    });

    it('calls remove item when delete button clicked', () => {
        wrapper.find('button').simulate('click');
        expect(props.handleRemoveOne).toHaveBeenCalledWith(props.id);
    });
});