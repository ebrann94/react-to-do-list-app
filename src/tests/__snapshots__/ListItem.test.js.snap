// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListItem component tests renders without crashing given required props 1`] = `ShallowWrapper {}`;
